### PR TITLE
pg-cursor, the read method now returns the generic type passed to the constructor

### PR DESCRIPTION
**Pull request by rafaelgilete**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Review by elias-fauser — APPROVED** [review]
Thanks for finding this bug. This was my fault. For a test you may just use a property on the typed return value

**Comment by iRafaelG** [comment]
Hello,

Thanks for the approval.

I don't really know how to do the test, but I used the callback test as a reference.

I have also changed the header version as indicated in the documentation, I forgot to do it the first time.

Best regards.

File: types/pg-cursor/pg-cursor-tests.ts [hunk 1]
```diff
@@ -47,9 +47,12 @@ const handleFn: ResultCallback<string> = (err: Error | undefined, rows: string[]
 handle.read(100, handleFn);
 
 // Returns promise when no callback
-handle.read(100).then(() => {
-    // Finished
-});
+handle
```
Comment (elias-fauser) [inline types/pg-cursor/pg-cursor-tests.ts:50]: To test the type more thorough you could type the cursor here with some arbitrary type with a property and the check that the returned rows are indeed properly typed:

```const cursor = new Cursor<{foo: string}>('SELECT * FROM table');
const handle = client.query(cursor);
cursor.read((rows) => {
rows.foo
})

File: types/pg-cursor/index.d.ts [hunk 2]
```diff
@@ -25,9 +25,9 @@ declare namespace Cursor {
 declare class Cursor<Row = any> extends EventEmitter {
     constructor(query: string, values?: any[], config?: Cursor.CursorQueryConfig);
     submit: (connection: Connection) => void;
-    read(maxRows: number): Promise<void>;
+    read(maxRows: number): Promise<Row>;
```
Comment (elias-fauser) [inline types/pg-cursor/index.d.ts:28]: This should rather be `Row[]` and then the type corresponds to a single row item

**Review by elias-fauser — CHANGES_REQUESTED** [review]

File: types/pg-cursor/index.d.ts (hunk 2 above)
Comment (iRafaelG) [reply types/pg-cursor/index.d.ts]: Fixed!

**Review by elias-fauser — APPROVED** [review]

**Pull request merged**